src/target/at32f43x.c

```diff
@@ -364,3 +393,164 @@ static bool at32f43_mass_erase(target_s *target)
 		return at32f43_mass_erase_bank(target, AT32F43x_FLASH_BANK2_REG_OFFSET, &timeout);
 	return true;
 }
+
+static bool at32f43_option_erase(target_s *target)
+{
+	/* bank_reg_offset is 0, option bytes belong to first bank */
+	at32f43_flash_clear_eop(target, 0);
+	DEBUG_TARGET("%s\n", __func__);
+
+	/* Wipe User System Data */
+	target_mem32_write32(target, AT32F43x_FLASH_CTRL, AT32F43x_FLASH_CTRL_USDERS | AT32F43x_FLASH_CTRL_USDULKS);
+	target_mem32_write32(target, AT32F43x_FLASH_CTRL,
+		AT32F43x_FLASH_CTRL_USDERS | AT32F43x_FLASH_CTRL_USDULKS | AT32F43x_FLASH_CTRL_ERSTR);
+
+	return at32f43_flash_busy_wait(target, 0, NULL);
+}
+
+static bool at32f43_option_write_erased(target_s *const target, const size_t offset, const uint16_t value)
+{
+	if (value == 0xffffU)
+		return true;
+
+	at32f43_flash_clear_eop(target, 0);
+
+	/* Enable writing User System Data */
+	target_mem32_write32(target, AT32F43x_FLASH_CTRL, AT32F43x_FLASH_CTRL_USDPRGM | AT32F43x_FLASH_CTRL_USDULKS);
+
+	const uint32_t addr = AT32F43x_USD_BASE + (offset * 2U);
+	DEBUG_TARGET("%s: 0x%08" PRIX32 " <- 0x%04X\n", __func__, addr, value);
+	const uint32_t time_start = platform_time_ms();
+	target_mem32_write16(target, addr, value);
+
+	const bool result = at32f43_flash_busy_wait(target, 0, NULL);
+	const uint32_t time_end = platform_time_ms();
+	const uint32_t time_spent = time_end - time_start;
+	if (time_spent > 20U)
+		DEBUG_TARGET("%s: took %" PRIu32 " ms\n", __func__, time_spent);
+	if (result || offset != 0U)
+		return result;
+
+	/* For error on offset 0, that is RDP byte, signal back the failure to erase RDP (?) */
+	const uint8_t status =
+		target_mem32_read32(target, AT32F43x_FLASH_STS) & (AT32F43x_FLASH_STS_PRGMERR | AT32F43x_FLASH_STS_EPPERR);
+	return status == AT32F43x_FLASH_STS_PRGMERR;
+}
+
+static bool at32f43_option_overwrite(target_s *const target, const uint16_t *const opt_val, const uint16_t ob_count)
+{
+	if (!at32f43_option_erase(target))
+		return false;
+
+	/* Write changed values using 16-bit accesses. */
+	for (size_t i = 0U; i < ob_count; ++i)
```

Comment:
The outer loop here should have braces added as, while this is legal, the `if` statement inside reads as a multi-line loop body and so the braces should be included in this situation.

Reply:
The rule per ~~src/target/HACKING.md~~ multiple PR feedback seems to be
```c
if (expr)
    statement1;
else
    statement2;
```
and
```c
for (;;)
    statement-iter;
```
but
```c
for (;;) {
    if (expr)
        statement1;
}
```
Maybe add this to the codestyle doc? Or there already is one? pre-commit/clang-format don't catch this, and I don't (yet) use clang-tidy.

Reply:
Correct. It's something where we started to deal with working through the kernel's and trying to write one but got stuck dealing with the super bombastic and kinda horrible way the kernel style guide is written so lost steam.. it's something we have on our list to deal with though as yeah, need a style guide.